### PR TITLE
Booking

### PR DESCRIPTION
**Pull request by Arash-SJPI22**
Has mock data that should be removed when PR #3 is merged.

### CODE REVIEW

File: src/pages/booking.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+{
+    // Mock data to be removed when real data is fixed
+    let title = "Blade"
+    let date = "2023-06-24"
+    let time = "14:30"
+    let salong = "Alcazar"
+    let lenght = "120"
+    let imageURL = "https://m.media-amazon.com/images/M/MV5BOTk2NDNjZWQtMGY0Mi00YTY2LWE5MzctMGRhZmNlYzljYTg5XkEyXkFqcGdeQXVyMTAyNjg4NjE0._V1_SX300.jpg"
+
+    const [adultPrice, setAdultPrice] = useState(150);
```
Comment (saeraa) [inline src/pages/booking.js]: I fixed the Link on the movie page.

To access the information, use the following:

```javascript
export function getServerSideProps(context) {
  console.log(context.query) 
  return {
      props: { 
         title: context.query.title,
         length: context.query.length,
         showtimeId: context.query.showtimeId // etc...
      }
  }
}```

**Review by davidlidebrandt — COMMENTED** [review]
I'm getting an error when trying to access /booking, am I missing something ? "Server Error
Error: Error serializing `.showtimeId` returned from `getServerSideProps` in "/booking".
Reason: `undefined` cannot be serialized as JSON. Please use `null` or omit this value.

This error happened while generating the page. Any console logs will be displayed in the terminal window."

File: src/pages/booking.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                            <p className={styles["child"]}>Barn / Pensionär ({childPrice}kr)</p>
+                            <div className={styles["adult-ticket"]}>
+                                <button className={styles["sub"]} onClick={() => setAdultTickets(checkInterval(adultTickets - 1) ? adultTickets - 1 : adultTickets)}>-</button>
+                                <input type="number" pattern='\d{2}' value={adultTickets} disabled />
+                                <button className={styles["add"]} onClick={() => setAdultTickets(checkInterval(adultTickets + 1) ? adultTickets + 1 : adultTickets)}>+</button>
+                            </div>
+                            <div className={styles["child-ticket"]}>
+                                <button className={styles["sub"]} onClick={() => setChildTickets(checkInterval(childTickets - 1) ? childTickets - 1 : childTickets)}>-</button>
+                                <input type="number" pattern='\d{2}' value={childTickets} disabled />
+                                <button className={styles["add"]} onClick={() => setChildTickets(checkInterval(childTickets + 1) ? childTickets + 1 : childTickets)}>+</button>
```
Comment (saeraa) [inline src/pages/booking.js]: There are some accessibility issues in your code here; the inputs should always have labels and the buttons should probably have an aria-label with ie "increase number of child tickets by 1".

File: src/pages/booking.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            poster: context.query.poster,
+            length: context.query.length
+        }
+    }
+}
+
+export default function Booking({ showtimeId, showtimeSalong, showtimedate, title, poster, length })
+{
+    const [adultPrice, setAdultPrice] = useState(150);
+    const [childPrice, setChildPrice] = useState(100);
```
Comment (saeraa) [inline src/pages/booking.js:21]: I'm wondering why you're using useState for these as there is nowhere in the code for changing the price - using the set-functions?

**Comment by Arash-SJPI22** [comment]
> I'm getting an error when trying to access /booking, am I missing something ? "Server Error Error: Error serializing `.showtimeId` returned from `getServerSideProps` in "/booking". Reason: `undefined` cannot be serialized as JSON. Please use `null` or omit this value.
> 
> This error happened while generating the page. Any console logs will be displayed in the terminal window."

I used mock data before to show the page properly, but now I use real data and that data comes from the page before which is the only way to get to the page. I will fix that in the future if there is time since users wouldn't go to booking page directly.

File: src/pages/booking.js (hunk 3 above)
Comment (Arash-SJPI22) [reply src/pages/booking.js:21]: I did it that way for future implementations so the price could be changed. Not relevant now, but I left them in anyway.

File: src/pages/booking.js (hunk 2 above)
Comment (Arash-SJPI22) [reply src/pages/booking.js]: Forgot, will fix !

**Review by saeraa — APPROVED** [review]
👌

**Review by davidlidebrandt — APPROVED** [review]

**Pull request merged**